src/main/java/uk/gov/companieshouse/insolvency/data/controller/InsolvencyController.java

```diff
@@ -50,4 +52,23 @@ public InsolvencyController(Logger logger, InsolvencyService insolvencyService)
         return ResponseEntity.status(HttpStatus.OK).build();
     }
 
+    /**
+     * Retrieve company insolvency information for a company number.
+     *
+     * @param  companyNumber  the company number for insolvency
+     * @return  {@link CompanyInsolvency} return company insolvency information
+     */
+    @GetMapping("/company/{company_number}/insolvency")
+    public ResponseEntity<CompanyInsolvency> insolvency(
+            @PathVariable("company_number") String companyNumber) {
+        logger.info(String.format(
+                "Retrieving company insolvency information for company number %s",
+                companyNumber));
+
+        CompanyInsolvency companyInsolvency = insolvencyService.retrieveCompanyInsolvency(
```

Comment:
You may have already done, if not please
1) Check if the data returned matching existing Perl endpoint
2) Make sure routes file updated for this url pattern